go/cmd/dolt/commands/push.go

```diff
@@ -204,6 +206,12 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
 					verr = bdr.Build()
 				} else if src == ref.EmptyBranchRef {
 					verr = deleteRemoteBranch(ctx, dest, remoteRef, dEnv.DoltDB, destDB, remote)
+				} else if apr.Contains(ForcePushFlag) {
+					verr = deleteRemoteBranch(ctx, dest, remoteRef, dEnv.DoltDB, destDB, remote)
```

Comment:
Is it worth adding a TODO or something? This should be first-class at the protocol level and the remote should handle the check...this suffers from dropping the remote ref before we get the new one up there...

Reply:
definitely